system/lib/libc/musl/src/time/nanosleep.c

```diff
@@ -3,15 +3,18 @@
 #include "libc.h"
 #if __EMSCRIPTEN__
 #include <errno.h>
+#include <emscripten/threading.h>
 #endif
 
 int nanosleep(const struct timespec *req, struct timespec *rem)
 {
 #if __EMSCRIPTEN__
-	// Without threads support, nanosleep cannot actually work except to do a
-	// silly busy-wait. FIXME with pthreads in standalone mode
-	errno = ENOSYS;
```

Comment:
Actually before we switch to buys waiting.. do you think we should just return ENOSYS?   i.e. do you agree with this comment?

Reply:
I don't think we had a great reason to return ENOSYS before. There is some risk of behavior changing for users, but it seems worth it for the code simplification and consistency.

I guess an argument could be made to do the reverse, return ENOSYS everywhere, but that seems riskier. I'm not sure though.